### ISSUE TITLE
Add FAQ about the project status to the README file

### ISSUE THREAD

**Issue by christian-roggia** [go-mgo/mgo#416]
I have seen many other questions like this and I was wondering myself of what is the current status of the project.

I'd like to sum up how the situation is from the resources available:

- Last commit is from 1st august (2016)
- Pull requests are not getting through
- Issues are closed when unrelated to the project but not solved when related
- Last release on labix.org was the 2015.12.06
- Mailing list is dead (dozens of questions without an answer)

I suggest adding a few lines in the README (a short FAQ) where it is explained if the project can be trusted for production, if the project will be updated (if yes, when) or if the project is dead/deprecated/no longer maintained.

**Comment by niemeyer** [comment]
Here is the short story for some context:

mgo was started independently and after a couple of years got some sponsorship from MongoDB, but I got slightly frustrated with a few issues and decided to break the formal ties while we were still in good grounds. That explains why activity reduced at that time, and yes, I have too much to do elsewhere to be able to sustain attention here on a continuous base.

This project is production quality, though, and I and many other people have projects that depend on it. In part, that's exactly the problem. So far I haven't found any other maintainers that were both interested and capable to contribute to the project on an unattended basis without risking breaking everybody's production systems.

If you (you the reader, not just the top poster) see emails unanswered in the list, PRs unaddressed, bugs reported, by all means _answer them._ That's exactly the problem. People only care about the project until they get their own personal issue addressed.

So my plan is to keep this project alive and working. But rather than someone that is always here, you'll see bursts of activity when I sit down for several days in a raw and get my mind back into this space. As I said in a few other places, I'm planning a new stable release for April.

I'm sorry. I would like for it to be different, but turns out the day only has so many hours.

**Comment by hexadecy** [comment]
I know that MongoDB staff did some fixes for mgo, I've created a PR #394 to link them all.
I'm sure don't want to merge PRs or write to this repo, but managing issues could be done by someone else with lower skill than you.
Cause there is a lot of issues/questions, it will be nice to have at least labels on it.
Labels like needs a PR, needs investigation, bug, question...

**Comment by niemeyer** [comment]
Thanks a lot @hexadecy. Even just organizing and commenting on the issues is already tremendously useful, as I can then sweep through them all during the update process and have a proper discussion to base decisions on,

**Comment by vivace-io** [comment]
I'd be happy to help sort through the issues and identify bugs as well! I'm probably not the best choice for API changes, but I can help handle issues and identify/fix some of the bugs that turn up. I could also help filter out PRs with changes that break the API, labeling accordingly, etc.

Just let me know!

**Comment by djensen47** [comment]
>  I'm planning a new stable release for April.

It's June. Any updates? Thanks.

**Comment by djensen47** [comment]
First of all, thanks for all the hard work on this project. It's wonderful and awesomely useful!

> I'm planning a new stable release for April.

It's June. Any updates? Thanks.

**Comment by djensen47** [comment]
@jyemin @craigwilson @ShaneHarvey @dgottlieb Sorry to post here but 10gen/mgo doesn't have Github Issues set up. Any change that 10gen would be willing to make their mgo fork a more "official" golang lib for MongoDB?

It appears that @niemeyer is super, super busy so it would be great to have a lib with an active community and support. Or at the very least, turn on issues for the 10gen/mgo?

Thanks everybody for all the hard work on everything!

**Comment by niemeyer** [comment]
Turning on issues elsewhere won't do much good unless there's a team actively working on it.

This is open source, so I can't prevent you from forking, but if you do that please change its name as well.

**Comment by djensen47** [comment]
> Turning on issues elsewhere won't do much good unless there's a team actively working on it.

It looks like the 10gen team has been actively working on their fork. If I'm not mistaken, I think all of those changes are also Pull Requests here.

> This is open source, so I can't prevent you from forking, but if you do that please change its name as well.

Yeah, that makes sense too. Cheers! 

I just bring this up because it looks like you're super busy and involved with a lot of other projects and it would be sad to see all this great work no longer maintained.

**Comment by niemeyer** [comment]
I realize and apologize for that. I wish I could just spend the next month working on that alone, but that's indeed not my current priority, so it keeps being pushed forward. Even then, we have important projects that depend on the stability of this driver, so I can't just open the gates.

If they want to fork, I honestly will understand. But this won't be mgo anymore, and they get to keep the whole pie.

**Comment by inancgumus** [comment]
@niemeyer I understand your concerns about breaking dependencies with the existing projects/teams that use mgo in their production-level projects. However, may I suggest that accepting PRs in a new mgo version to prevent breaking the existing projects?

**Comment by niemeyer** [comment]
@inancgumus As mentioned above, definitely happy to see the community supporting the project, but not so happy about letting it go loose. During the years I supported the project very actively, we couldn't get someone to really help maintain the project. To be clear, I'm not complaining &ndash; we got many excellent contributions from the community and from MongoDB itself and those got integrated, but we don't have a second maintainer yet, and that means there was nobody with the right profile and a real long term interest to engage.

I'm working on that, and hope to have a proper solution to it soon.

**Comment by djensen47** [comment]
A new maintainer? That would be nice. 😎 🌈 🦄

**Comment by niemeyer** [comment]
Yeah, it would be **excellent** really!  That's true for any of the projects I'm responsible for, by the way. Even for the ones I'm actively working on, long term maintenance from responsible and experienced developers is most appreciated, by all means.

It's also somewhat hard to find, unfortunately. Such maintainers in general join organically, and stick around because they care in the first place, instead of simply acquiring a maintainer label because someone says so. They **are** maintainers, instead of _becoming_ a maintainer. I've seen the latter case fail repeatedly.

**Comment by fmpwizard** [comment]
Maybe this will be better for the new maintainer or maybe you'll have enough time but I submitted a small pr that will fix  all the false failures from Travis https://github.com/go-mgo/mgo/pull/462 including my other pr. Happy mgo user here who also looks forward to the news

**Comment by hexadecy** [comment]
@niemeyer you suggested that we change the name if we fork and create a new issue tracker?
mango, moongo, mgo-plus, mgo.v3, mgo-next, mgoo, mogun, mgo-alt

**Comment by vivace-io** [comment]
I would love to work with someone to maintain the project. Not sure I could do it _all_ alone, but I know I can at least maintain the issue tracker, close out duplicates, etc. It would be very nice to have the chance to work as part of a team in a project like this. 

I have no doubt we could kick it back in to gear easily.  😃

**Comment by davidrenne** [comment]
I understand @niemeyer your situation with this library.  I wonder if there is anyone who has actively forked this already who should jump into the conversation for this potential fork.  I still am curious if 10gen has a version, where is it and does it have some changes we can compare to master of this source repo?

With a year gap in modifying this repository of any bug fixes, I am thinking for our production release we need to always stick on this version and never anything else new that could be breaking.

**Comment by domodwyer** [comment]
@davidrenne we (@globalsign) are maintaining a fork of mgo that we use internally - it's mostly aimed at bugfixes but there's also been feature additions from both ourselves and 3rd parties. We're committed to keeping it alive and updated for the long term. 

We merge PRs into development, which is then tested in a sharded, replicated environment in our datacenter for regressions. Once everyone is happy, we merge to master - this is to maintain a bit of quality control past the usual PR process.

**Comment by inancgumus** [comment]
I changed my `go-mgo/mgo` to `globalsign/mgo` then run my project's integration tests, everything seems fine and working.

Although there are many [changes made](https://github.com/go-mgo/mgo/compare/v2...globalsign:master) to `go-mgo/mgo` on @domodwyer 's fork; it works flawlessly on my own project.
